simulator/src/scrypto/cmd_coverage.rs

```diff
@@ -0,0 +1,204 @@
+use clap::Parser;
+use radix_engine_interface::types::Level;
+use regex::Regex;
+use std::env;
+use std::env::current_dir;
+use std::fs;
+use std::io::Read;
+use std::io::Write;
+use std::path::PathBuf;
+use std::process::Command;
+use walkdir::WalkDir;
+
+use crate::scrypto::*;
+use crate::utils::*;
+
+/// Run Scrypto tests and generate code coverage report
+#[derive(Parser, Debug)]
+pub struct Coverage {
+    /// The arguments to be passed to the test executable
+    arguments: Vec<String>,
+
+    /// The package directory
+    #[clap(long)]
+    path: Option<PathBuf>,
+}
+
+impl Coverage {
+    fn check_command_availability(command: String) -> Result<(), Error> {
+        if Command::new(&command).arg("--version").output().is_err() {
+            println!("Missing command: {}. Please install LLVM version matching rustc LLVM version, which is {}.", 
+                command, command.split('-').last().unwrap_or("Unknown"));
+            println!("For more information, check https://apt.llvm.org/");
+            return Err(Error::CoverageError(CoverageError::MissingLLVM));
+        }
+        Ok(())
+    }
+
+    pub fn run(&self) -> Result<(), Error> {
+        let output = Command::new("rustc")
+            .arg("--version")
+            .arg("--verbose")
+            .output()
+            .expect("Failed to execute rustc command");
+
+        let output_str = String::from_utf8(output.stdout).expect("Failed to read rustc output");
+        let is_nightly = output_str.contains("nightly");
+        let llvm_major_version = Regex::new(r"LLVM version: (\d+)")
+            .unwrap()
+            .captures(&output_str)
+            .and_then(|cap| cap.get(1).map(|m| m.as_str()))
+            .expect("Failed to read LLVM version of rustc");
+
+        if !is_nightly {
+            println!("Coverage tool requries nightly version of rust toolchain");
+            println!("You can install it by using the following commands:");
+            println!("rustup default nightly");
```

Comment:
Wonder if it's possible to use `nightly` for coverage only?

Reply:
Yes, it could use "cargo +nightly" when building package, but there may be some issues when someone is using multiple nightly versions